sdk/search/Azure.Search.Documents/src/Indexes/Models/VectorSearchField.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Search.Documents.Indexes.Models
+{
+    /// <summary>
+    /// A searchable vector field of type "Collection(<see cref="SearchFieldDataType.Single"/>)".
+    /// </summary>
+    public class VectorSearchField : SearchFieldTemplate
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SimpleField"/> class.
+        /// </summary>
+        /// <param name="name">The name of the field, which must be unique within the index or parent field.</param>
+        /// <param name="vectorSearchDimensions">The dimensionality of the vector field.</param>
+        /// <param name="vectorSearchProfileName">The name of the vector search profile that specifies the algorithm to use when searching the vector field.</param>
+        /// <exception cref="ArgumentException"><paramref name="name"/> is an empty string.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/> is null.</exception>
+        public VectorSearchField(string name, int vectorSearchDimensions, string vectorSearchProfileName) : base(name, SearchFieldDataType.Collection(SearchFieldDataType.Single))
+        {
+            VectorSearchDimensions = vectorSearchDimensions;
+            VectorSearchProfileName = vectorSearchProfileName;
+        }
+
+        /// <summary> The dimensionality of the vector field. </summary>
+        public int VectorSearchDimensions { get; set; }
+
+        /// <summary> The name of the vector search profile that specifies the algorithm to use when searching the vector field. </summary>
+        public string VectorSearchProfileName { get; set; }
```

Comment:
@alzimmermsft / @tg-msft Since we've introduced a separate type for vector fields called `VectorSearchField`, does it make sense to rename the properties as follows:
- `VectorSearchDimensions` to `Dimensions`
- `VectorSearchProfileName` to `ProfileName`

We can keep the property names `VectorSearchDimensions` and `VectorSearchProfileName` in the `SearchField`.

Reply:
Similarly for `VectorSearchFieldAttribute` type.

Reply:
This is a really tough one.  Normally I'd say shorten them like you propose.  But in this case, the only use is for defining an index where we've basically got properties like this referring to different parts in a massive object graph.  I think the more verbose names actually help people glue these concepts together.  I'd vote for leaving them.  (I could be talked out of that position though if anyone feels really strongly about this. 😄).